src/tests/integration/character-countdown.test.ts

```diff
@@ -0,0 +1,154 @@
+import { fromText, getPlainText } from '../../draft-js-utils'
+
+// TODO: Hank to remove this .only when done writing these tests
+describe.only('character countdown', () => {
+  const mocks = createMocks()
+
+  runBackground(mocks, { alreadyAuthenticated: true })
+  mountPopup(mocks, { alreadyAuthenticated: true, handle: 'exists' })
+
+  describe('when there are many characters remaining', () => {
+    it("renders a progress circle that is filled in according to Twitter's 280 character limit", () => {
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio + filledInRatio).to.closeTo(1, 0.00000000000001)
+    })
+
+    it('renders no text', () => {
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const text = characterCountdown.querySelector('svg > text')
+      expect(text).to.equal(null)
+    })
+
+    // TODO: Hank to fix bug in code
+    it('is a progressbar with the an aria-valuenow equal to the filledInRatio', () => {
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+
+      expect(Number(characterCountdown.getAttribute('aria-valuenow'))).to.equal(Math.round(filledInRatio * 100))
+    })
+
+    it('does not disable posting', () => {
+      const postButton = mocks.app().querySelector('.post-btn')! as HTMLButtonElement
+      expect(postButton.disabled).to.equal(false)
+    })
+  })
+
+  describe('when approaching the character limit', () => {
+    const almost280Chars = '@zing.com ' + 'x'.repeat(260)
+    before(() => {
+      mocks.backgroundWindow.store.dispatchers.updateEditorState({
+        editorState: fromText(almost280Chars),
+      })
+    })
+
+    it('renders a more full progress circle', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio).to.closeTo(0, 0.1)
```

Comment:
Make these `0.00000000000001` as 90% is not close enough to 100% for our liking.

Reply:
syntax changed and meet the bar.